### PR TITLE
Bug 1210697 - [smart-system] Add SyncManager and SyncStateMachine. r=…

### PR DESCRIPTION
**Pull request by ferjm**
…yifan

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=867997f9c13f7ab0e3834efe24a0c20d202b1f9d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fe538fd5eccf22bd094af203307b25b22d4234a7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d83a6245ae9d58db8fb195951fa1d347d2022c03

File: tv_apps/smart-system/js/sync_state_machine.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  for (var name in transitions) {
+    if (!transitions.hasOwnProperty(name)) {
+      continue;
+    }
+    SyncStateMachine[name] = createTransition(name, transitions[name]);
+  }
+
+  // Exported only for testing purposes.
+  exports.SyncStateMachine = SyncStateMachine;
```
Comment (begeeben) [inline tv_apps/smart-system/js/sync_state_machine.js]: Guess this line should be added above the comment since it's needed in SyncManager?

Comment (ferjm) [inline tv_apps/smart-system/js/sync_state_machine.js]: Yes!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c579eb44d477d5d39fd9097897c402d68a3c6209

**Pull request merged**